Controllers/PurchasesController.cs

```diff
@@ -81,8 +89,15 @@ public async Task<IActionResult> Details(int? id)
                 .Include(b => b.Branch)
                 .Include(b => b.Branch.Entity)
                 .Include(b => b.Customer)
+                .Include(b => b.PurchaseItems)
```

Comment:
I guess foreach can be reduced(all loading work in query) to query part:
https://docs.microsoft.com/pl-pl/ef/core/querying/related-data

look espetially at this:
`    var blogs = context.Blogs
        .Include(blog => blog.Posts)
            .ThenInclude(post => post.Author)
        .ToList();`

Reply:
Applied.